config/interpolate_funcs.go

```diff
@@ -640,6 +641,44 @@ func appendIfMissing(slice []string, element string) []string {
 	return append(slice, element)
 }
 
+// interpolationFuncFilter implements the "filter" function that
+// removes elements from a list
+func interpolationFuncFilter() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeList, ast.TypeList, ast.TypeString},
+		ReturnType: ast.TypeList,
+		Callback: func(args []interface{}) (interface{}, error) {
+			var list []string
+
+			argument, _ := args[0].([]ast.Variable)
+			keys, _ := args[1].([]ast.Variable)
+			searchkey, _ := args[2].(string)
```

Comment:
Type checking in this case is enforced by ast.Functions's ArgType, so if user put a wrong argument, he would get an error like that:

```
* google_compute_instance_group.kube_minion[0]: At column 3, line 1: filter: argument 1 should be type list, got type string in:

${filter("test",
      google_compute_instance.kube_minion.*.zone,
      element(var.cluster_zones, count.index))}
```